pkgs/build-support/go/module.nix

```diff
@@ -6,392 +6,390 @@
-          setSourceRoot = finalAttrs.setSourceRoot or "";
-          env = finalAttrs.env or { };
-
-          impureEnvVars = lib.fetchers.proxyImpureEnvVars ++ [
-            "GIT_PROXY_COMMAND"
-            "SOCKS_SERVER"
-            "GOPROXY"
-          ];
-
-          configurePhase =
-            args.modConfigurePhase or ''
-              runHook preConfigure
-              export GOCACHE=$TMPDIR/go-cache
-              export GOPATH="$TMPDIR/go"
-              cd "$modRoot"
-              runHook postConfigure
-            '';
-
-          buildPhase =
-            args.modBuildPhase or (
-              ''
-                runHook preBuild
-              ''
-              + lib.optionalString finalAttrs.deleteVendor ''
-                if [ ! -d vendor ]; then
-                  echo "vendor folder does not exist, 'deleteVendor' is not needed"
-                  exit 10
-                else
-                  rm -rf vendor
-                fi
-              ''
-              + ''
-                if [ -d vendor ]; then
-                  echo "vendor folder exists, please set 'vendorHash = null;' in your expression"
-                  exit 10
-                fi
+  extendDrvArgs =
+    finalAttrs:
+    {
+      name ? "${args'.pname}-${args'.version}",
+      # The source used to build the derivation.
+      src,
+      # Native build inputs used for the derivation.
+      nativeBuildInputs ? [ ],
+      passthru ? { },
+      patches ? [ ],
+
+      # A function to override the `goModules` derivation.
+      overrideModAttrs ? (finalAttrs: previousAttrs: { }),
+
+      # Directory to the `go.mod` and `go.sum` relative to the `src`.
+      modRoot ? "./",
+
+      # The SRI hash of the vendored dependencies.
+      # If `vendorHash` is `null`, no dependencies are fetched and
+      # the build relies on the vendor folder within the source.
+      vendorHash ? throw (
+        if args' ? vendorSha256 then
+          "buildGoModule: Expect vendorHash instead of vendorSha256"
+        else
+          "buildGoModule: vendorHash is missing"
+      ),
+
+      # Whether to delete the vendor folder supplied with the source.
+      deleteVendor ? false,
+
+      # Whether to fetch (go mod download) and proxy the vendor directory.
+      # This is useful if your code depends on c code and go mod tidy does not
+      # include the needed sources to build or if any dependency has case-insensitive
+      # conflicts which will produce platform dependant `vendorHash` checksums.
+      proxyVendor ? false,
+
+      # We want parallel builds by default.
+      enableParallelBuilding ? true,
+
+      # Do not enable this without good reason
+      # IE: programs coupled with the compiler.
+      allowGoReference ? false,
+
+      # Meta data for the final derivation.
+      meta ? { },
+
+      # Go linker flags.
+      ldflags ? [ ],
+      # Go build flags.
+      GOFLAGS ? [ ],
+
+      ...
+    }@args':
```

Comment:
How about changing this from `args'` to `args`? It does change the semantics, but not much.

Reply:
Done